halo2_proofs/src/poly/evaluator.rs

```diff
@@ -442,10 +500,12 @@ impl BasisOps for Coeff {
 
     fn linear_term<F: FieldExt>(
         _: &EvaluationDomain<F>,
+        poly_len: usize,
         chunk_size: usize,
         chunk_index: usize,
         scalar: F,
     ) -> Vec<F> {
+        let chunk_size = cmp::min(chunk_size, poly_len - chunk_size * chunk_index);
         let mut chunk = vec![F::zero(); chunk_size];
```

Comment:
Sorry, the change I suggested we make here is actually wrong. `chunk_size` on line 513 should be the *original* chunk size, because if `chunk_index == 1` then what we also want to check is whether chunk 0 (not chunk 1) was of size 1.

```suggestion
        let mut chunk = vec![F::zero(); cmp::min(chunk_size, poly_len - chunk_size * chunk_index)];
```

Also maybe add to the comment to explain this.